### PR TITLE
Add io.itch.ksylvestre.MightyGrepDemo

### PR DESCRIPTION
**Pull request by kyle-sylvestre**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*

1. --filesystem=host: allow users to search more files on their system
2. --talk-name=org.freedesktop.Flatpak: allow users to use specify "Open in editor" and "Open file location" commands

- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by kyle-sylvestre** [comment]
applied for finish-args-flatpak-spawn-access [here](https://github.com/flathub-infra/flatpak-builder-lint/pull/242)

File: dev.ksylvestre.MightyGrepDemo.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    <screenshot type="default">
+      <image type="remote" height="758" width="1284">https://raw.githubusercontent.com/flathub/dev.ksylvestre.MightyGrepDemo/master/dev.ksylvestre.MightyGrepDemo-dark.png</image>
+      <image type="remote" height="758" width="1284">https://raw.githubusercontent.com/flathub/dev.ksylvestre.MightyGrepDemo/master/dev.ksylvestre.MightyGrepDemo-light.png</image>
+    </screenshot>
+  </screenshots>
+  <categories>
+    <category>Development</category>
+    <category>Utility</category>
+  </categories>
+  <icon type="remote" height="256" width="256">https://raw.githubusercontent.com/flathub/dev.ksylvestre.MightyGrepDemo/master/dev.ksylvestre.MightyGrepDemo-256.png</icon>
```
Comment (hfiguiere) [inline dev.ksylvestre.MightyGrepDemo.appdata.xml:32]:
```suggestion
```

don't use remote icons.

File: dev.ksylvestre.MightyGrepDemo.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      <li>Jump to match in external editor</li>
+      <li>Navigate search history with side mouse buttons</li>
+      <li>Built-in file viewer</li>
+      <li>Export search results to clipboard, file, or pipe</li>
+    </ul>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image type="remote" height="758" width="1284">https://raw.githubusercontent.com/flathub/dev.ksylvestre.MightyGrepDemo/master/dev.ksylvestre.MightyGrepDemo-dark.png</image>
+      <image type="remote" height="758" width="1284">https://raw.githubusercontent.com/flathub/dev.ksylvestre.MightyGrepDemo/master/dev.ksylvestre.MightyGrepDemo-light.png</image>
```
Comment (hfiguiere) [inline dev.ksylvestre.MightyGrepDemo.appdata.xml:25]: these files don't exist, the build WILL fail.

File: dev.ksylvestre.MightyGrepDemo.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>dev.ksylvestre.MightyGrepDemo</id>
```
Comment (hfiguiere) [inline dev.ksylvestre.MightyGrepDemo.appdata.xml:3]: `ksylvestre.dev` doesn't exist. the application id doesn't meet the requirements.

File: dev.ksylvestre.MightyGrepDemo-256.png
Comment (hfiguiere) [inline dev.ksylvestre.MightyGrepDemo-256.png:1]: this file should be part of the upstream distribution

File: dev.ksylvestre.MightyGrepDemo.appdata.xml
Comment (hfiguiere) [inline dev.ksylvestre.MightyGrepDemo.appdata.xml:1]: this file should be part of the upstream distribution

File: dev.ksylvestre.MightyGrepDemo.desktop
Comment (hfiguiere) [inline dev.ksylvestre.MightyGrepDemo.desktop:1]: this file should be part of the upstream distribution

File: flathub.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: I won't state how disappointing that is.

File: dev.ksylvestre.MightyGrepDemo-dark.png
Comment (hfiguiere) [inline dev.ksylvestre.MightyGrepDemo-dark.png:1]: screenshot should be hosted upstream since you are upstream.

**Comment by kyle-sylvestre** [comment]
remaking with proper branch name

**Pull request closed without merging**